services/requests/request_states/schemas.go

```diff
@@ -0,0 +1,77 @@
+package request_states
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+)
+
+var nameRegex = regexp.MustCompile(`^[A-Z](?:[A-Z_]*[A-Z])?$`)
```

Comment:
It seems to be the same everywhere, maybe you can factorize it and put it in a helper?

Reply:
I did it, is the solution good ?

Reply:
I just meant to move nameRegex in a package and make it public (by capitalizing the first name) so you could have kept `validation.StringMatch(NameRegex, "Must be a valid State name")` but you're solution also works

Reply:
it seemed easier to use for the user